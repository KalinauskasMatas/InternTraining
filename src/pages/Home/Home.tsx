import React from "react";
import Footer from "../../components/Footer/Footer";
import Header from "../../components/Header/Header";
import MoviesList from "../../components/MoviesList/MoviesList";
import NavBar from "../../components/NavBar/NavBar";

const Home = () => {
  return (
    <div className="body-wrapper">
      <Header title={"Movie rental"} />
      <NavBar />
      <MoviesList />
      <Footer />
    </div>
  );
};

export default Home;
