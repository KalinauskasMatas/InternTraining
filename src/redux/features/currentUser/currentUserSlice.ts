import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import { RootState } from "../../store";
import { UserState } from "../../interfaces";

const initialState: UserState =
  JSON.parse(localStorage.getItem("currUser") as string) || {};

const currentUserSlice = createSlice({
  name: "currentUser",
  initialState,
  reducers: {
    setUser: (state, action: PayloadAction<UserState>) => {
      state.fname = action.payload.fname;
      state.surname = action.payload.surname;
      state.email = action.payload.email;
      state.password = action.payload.password;
      localStorage.setItem("currUser", JSON.stringify(state));
    },
    clearUser: (state) => {
      localStorage.removeItem("currUser");
      state = { ...initialState };
      console.log(state);
    },
  },
});

export const { setUser, clearUser } = currentUserSlice.actions;

export const selectCurrUser = (state: RootState) => state.currentUser;

export default currentUserSlice.reducer;
