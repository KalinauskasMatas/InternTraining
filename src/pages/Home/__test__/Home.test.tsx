import Home from "../Home";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";

import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { configureStore } from "@reduxjs/toolkit";
import currentUserSlice from "../../../redux/features/currentUser/currentUserSlice";

import { BrowserRouter } from "react-router-dom";

const store = configureStore({
  reducer: {
    currentUser: currentUserSlice,
  },
});

afterEach(cleanup);

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    </Provider>
  );
});

it("matches with snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
