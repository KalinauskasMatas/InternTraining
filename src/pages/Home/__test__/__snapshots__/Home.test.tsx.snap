// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches with snapshot 1`] = `
<div
  className="body-wrapper"
>
  <header
    className="header-name"
    data-testid="header-title"
  >
    Movie rental
  </header>
  <nav>
    <ul>
      <span
        className="left"
      >
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/yourMovies"
            onClick={[Function]}
          >
            Your movies
          </a>
        </li>
        <li
          className="profile-link"
        >
          <a
            href="/profile"
            onClick={[Function]}
          >
            Profile
          </a>
        </li>
      </span>
      <span>
        <li
          id="logout"
          onClick={[Function]}
        >
          Logout
        </li>
      </span>
    </ul>
  </nav>
  <section
    className="available-movies"
  >
    <h2
      className="available-movies-header"
    >
      Available Movies
    </h2>
    <table
      id="available-movies-table"
    >
      <tbody>
        <tr>
          <th>
            Name
          </th>
          <th>
            Genre
          </th>
          <th>
            Price for 12h
          </th>
          <th>
            Is in stock
          </th>
        </tr>
      </tbody>
    </table>
  </section>
  <footer>
    <span>
      Made by: Matas Kalinauskas
    </span>
    <span>
      2023
    </span>
  </footer>
</div>
`;
